lib/symboldatabase.cpp

```diff
@@ -4427,8 +4427,11 @@ const Function * Function::getOverriddenFunctionRecursive(const ::Type* baseType
 
 const Variable* Function::getArgumentVar(nonneg int num) const
 {
-    if (num < argumentList.size())
-        return &argumentList[num];
+    if (num < argumentList.size()) {
+        auto it = argumentList.begin();
+        std::advance(it, num);
```

Comment:
I don't understand.. is this safer or faster in some way?

Reply:
How would you do it with a `std::list`?

Reply:
ok understood.. it seems unfortunate to use a `std::list` but not sure how to solve it properly with the vector. :-(